paymentsheet/src/test/java/com/stripe/android/paymentsheet/repositories/CustomerRepositoryKtxTest.kt

```diff
@@ -0,0 +1,206 @@
+package com.stripe.android.paymentsheet.repositories
+
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.testing.PaymentMethodFactory
+import com.stripe.android.utils.FakeCustomerRepository
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+import org.mockito.Mockito.spy
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.verify
+
+class CustomerRepositoryKtxTest {
+    @Test
+    fun `'detachPaymentMethodAndDuplicates' attempts to remove all duplicate cards with matching fingerprints`() =
+        runTest {
+            val fingerprint = "4343434343"
+            val differentFingerprint = "5454545454"
+
+            val cards = PaymentMethodFactory.cards(size = 5)
+            val cardsWithFingerprints = cards.mapIndexed { index, paymentMethod ->
+                if (index < 3) {
+                    paymentMethod.copy(
+                        card = paymentMethod.card?.copy(
+                            fingerprint = fingerprint
+                        )
+                    )
+                } else {
+                    paymentMethod.copy(
+                        card = paymentMethod.card?.copy(
+                            fingerprint = differentFingerprint
+                        )
+                    )
+                }
+            }
+
+            val repository = spy(
+                FakeCustomerRepository(
+                    paymentMethods = cardsWithFingerprints,
```

Comment:
rather than using a spy here, can we update the fake to expose what you're trying to test?